packages/sanity/src/core/releases/tool/detail/events/buildReleaseEditEvents.ts

```diff
@@ -0,0 +1,53 @@
+import {type TransactionLogEventWithEffects} from '@sanity/types'
+
+import {applyMendozaPatch} from '../../../../preview/utils/applyMendozaPatch'
+import {type ReleaseDocument, type ReleaseType} from '../../../store/types'
+import {getReleaseIdFromReleaseDocumentId} from '../../../util/getReleaseIdFromReleaseDocumentId'
+import {type CreateReleaseEvent, type EditReleaseEvent} from './types'
+
+export function buildReleaseEditEvents(
+  transactions: TransactionLogEventWithEffects[],
+  release: ReleaseDocument,
+): (EditReleaseEvent | CreateReleaseEvent)[] {
+  // Confirm we have all the events by checking the first transaction id and the release._rev, the should match.
+  if (release._rev !== transactions[0]?.id) {
+    console.error('Some transactions are missing, cannot calculate the edit events')
```

Comment:
In the way we are using it we should never get into this state, because we always get the latest transactions before going through here.
https://github.com/sanity-io/sanity/pull/7973/files#diff-b8553af0c7faf96db130e45d99244d78f2a80247699054f6f4fdafa17f6b2d52R138-R151

Added this JIC we want to use this from other place and as a dev warning, the errors received from the applyPatch are not indicative of what is going on.